controlplane/kubeadm/OWNERS

```diff
@@ -0,0 +1,8 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+approvers:
+  - cluster-api-controlplane-provider-kubeadm-maintainers
+
+reviewers:
+  - cluster-api-reviewers
```

Comment:
I think you don't have to add `cluster-api-reviewers` again if you're not using `no_parent_owners`. But if it's preferred to make it more obvious it's of course fine to keep it.